ext/mysql2/violite.h

```diff
@@ -0,0 +1,280 @@
+/* Copyright (c) 2000, 2015, Oracle and/or its affiliates. All rights
```

Comment:
the other possibility here is to declare dependency on the two functions that the `has_next` pointer could point to, `vio_buff_has_data` or `vio_ssl_has_data`, but this would mean figuring out whether the underlying connection has negotiated an SSL connection or not, and it's not clear to me if we can figure that out.